crates/sui-indexer-alt/src/handlers/kv_epoch_starts.rs

```diff
@@ -72,4 +73,12 @@ impl Handler for KvEpochStarts {
             .execute(conn)
             .await?)
     }
+
+    async fn prune(range: PrunableRange, conn: &mut db::Connection<'_>) -> Result<usize> {
+        let (from, to) = range.containing_epochs();
+        let filter = kv_epoch_starts::table
+            .filter(kv_epoch_starts::epoch.between(from as i64, to as i64 - 1));
```

Comment:
I have not fully convinced myself that pruning epoch grained tables will just work like this, even when our watermark and retention are checkpoint grained. I will add a test tomorrow to convince myself.

Reply:
Indeed -- and I would say that if this implementation doesn't behave correctly, it would be good to change the helper function in `PrunableRange` so that all prune impls can follow the same pattern (regardless of whether they are epoch-, checkpoint-, or transaction-grained), rather than change the epoch-grained impls to have a slightly different structure.